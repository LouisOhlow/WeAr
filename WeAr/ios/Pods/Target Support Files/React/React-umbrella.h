#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "React/Base/RCTAssert.h"
#import "React/Base/RCTBridge.h"
#import "React/Base/RCTBridgeDelegate.h"
#import "React/Base/RCTBridgeModule.h"
#import "React/Base/RCTCache.h"
#import "React/Base/RCTConvert.h"
#import "React/Base/RCTDefines.h"
#import "React/Base/RCTEventDispatcher.h"
#import "React/Base/RCTFPSGraph.h"
#import "React/Base/RCTFrameUpdate.h"
#import "React/Base/RCTInvalidating.h"
#import "React/Base/RCTJavaScriptExecutor.h"
#import "React/Base/RCTJavaScriptLoader.h"
#import "React/Base/RCTJSMethodRegistrar.h"
#import "React/Base/RCTKeyboardObserver.h"
#import "React/Base/RCTKeyCommands.h"
#import "React/Base/RCTLog.h"
#import "React/Base/RCTModuleData.h"
#import "React/Base/RCTModuleMap.h"
#import "React/Base/RCTModuleMethod.h"
#import "React/Base/RCTPerformanceLogger.h"
#import "React/Base/RCTPerfStats.h"
#import "React/Base/RCTProfile.h"
#import "React/Base/RCTRootView.h"
#import "React/Base/RCTSparseArray.h"
#import "React/Base/RCTTouchHandler.h"
#import "React/Base/RCTURLRequestDelegate.h"
#import "React/Base/RCTURLRequestHandler.h"
#import "React/Base/RCTUtils.h"
#import "React/Executors/RCTContextExecutor.h"
#import "React/Executors/RCTWebViewExecutor.h"
#import "React/Layout/Layout.h"
#import "React/Modules/RCTAccessibilityManager.h"
#import "React/Modules/RCTAlertManager.h"
#import "React/Modules/RCTAppState.h"
#import "React/Modules/RCTAsyncLocalStorage.h"
#import "React/Modules/RCTDevLoadingView.h"
#import "React/Modules/RCTDevMenu.h"
#import "React/Modules/RCTExceptionsManager.h"
#import "React/Modules/RCTPointAnnotation.h"
#import "React/Modules/RCTRedBox.h"
#import "React/Modules/RCTSourceCode.h"
#import "React/Modules/RCTStatusBarManager.h"
#import "React/Modules/RCTTiming.h"
#import "React/Modules/RCTUIManager.h"
#import "React/Views/RCTActivityIndicatorViewManager.h"
#import "React/Views/RCTAnimationType.h"
#import "React/Views/RCTAutoInsetsProtocol.h"
#import "React/Views/RCTBorderDrawing.h"
#import "React/Views/RCTComponent.h"
#import "React/Views/RCTComponentData.h"
#import "React/Views/RCTConvert+CoreLocation.h"
#import "React/Views/RCTConvert+MapKit.h"
#import "React/Views/RCTDatePickerManager.h"
#import "React/Views/RCTMap.h"
#import "React/Views/RCTMapManager.h"
#import "React/Views/RCTModalHostView.h"
#import "React/Views/RCTModalHostViewController.h"
#import "React/Views/RCTModalHostViewManager.h"
#import "React/Views/RCTNavigator.h"
#import "React/Views/RCTNavigatorManager.h"
#import "React/Views/RCTNavItem.h"
#import "React/Views/RCTNavItemManager.h"
#import "React/Views/RCTPicker.h"
#import "React/Views/RCTPickerManager.h"
#import "React/Views/RCTPointerEvents.h"
#import "React/Views/RCTProgressViewManager.h"
#import "React/Views/RCTScrollableProtocol.h"
#import "React/Views/RCTScrollView.h"
#import "React/Views/RCTScrollViewManager.h"
#import "React/Views/RCTSegmentedControl.h"
#import "React/Views/RCTSegmentedControlManager.h"
#import "React/Views/RCTShadowView.h"
#import "React/Views/RCTSlider.h"
#import "React/Views/RCTSliderManager.h"
#import "React/Views/RCTSwitch.h"
#import "React/Views/RCTSwitchManager.h"
#import "React/Views/RCTTabBar.h"
#import "React/Views/RCTTabBarItem.h"
#import "React/Views/RCTTabBarItemManager.h"
#import "React/Views/RCTTabBarManager.h"
#import "React/Views/RCTTextDecorationLineType.h"
#import "React/Views/RCTView.h"
#import "React/Views/RCTViewControllerProtocol.h"
#import "React/Views/RCTViewManager.h"
#import "React/Views/RCTWebView.h"
#import "React/Views/RCTWebViewManager.h"
#import "React/Views/RCTWrapperViewController.h"
#import "React/Views/UIView+Private.h"
#import "React/Views/UIView+React.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

