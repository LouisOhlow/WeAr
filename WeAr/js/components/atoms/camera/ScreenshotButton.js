import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

export default function ScreenshotButton({ capturePhoto }) {
  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity style={styles.button} onPress={capturePhoto} />
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    borderWidth: 3,
    borderColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    width: 70,
    height: 70,
    backgroundColor: 'rgba(0,0,0,0.25)',
    borderRadius: 50,
  },
  buttonContainer: {
    position: 'absolute',
    bottom: 50,
    alignSelf: 'center',
  },
});
