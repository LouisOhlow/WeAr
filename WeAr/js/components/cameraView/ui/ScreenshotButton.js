import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import COLORS from '../../../res/colors';

/**
 * The Button which handles the video and photo reaction
 *
 * @param {function} capturePhoto is called when the button is pressed
 * @param {function} startVideo is called when the button is pressed and hold down
 * @param {function} stopVideo is called when the button was pressed long and then released
 */
export default function ScreenshotButton({ capturePhoto, startVideo, stopVideo }) {
  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity
        style={styles.button}
        onPress={capturePhoto}
        onLongPress={startVideo}
        onPressOut={stopVideo}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    borderWidth: 3,
    borderColor: COLORS.white,
    alignItems: 'center',
    justifyContent: 'center',
    width: 70,
    height: 70,
    backgroundColor: 'rgba(0,0,0,0.25)',
    borderRadius: 50,
  },
  buttonContainer: {
    position: 'absolute',
    bottom: 50,
    alignSelf: 'center',
  },
});
