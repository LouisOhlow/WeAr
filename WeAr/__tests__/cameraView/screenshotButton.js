import React from 'react';
import renderer from 'react-test-renderer';
import ScreenshotButton from '../../js/components/atoms/camera/ScreenshotButton';

describe('Screenshotbutton', () => {

    it('test PermissionHint Snapshot with blocked permission', () => {
        const tree = renderer
          .create(<ScreenshotButton />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
  });
  