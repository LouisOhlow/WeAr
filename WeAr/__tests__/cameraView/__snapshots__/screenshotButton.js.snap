// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screenshotbutton test PermissionHint Snapshot with blocked permission 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "bottom": 50,
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.25)",
        "borderColor": "#fff",
        "borderRadius": 50,
        "borderWidth": 3,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "width": 70,
      }
    }
  />
</View>
`;
